debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/SnapshotReader.java

```diff
@@ -659,29 +659,39 @@ protected void execute() {
     }
 
     protected void readBinlogPosition(int step, SourceInfo source, JdbcConnection mysql, AtomicReference<String> sql) throws SQLException {
-        logger.info("Step {}: read binlog position of MySQL master", step);
-        String showMasterStmt = "SHOW MASTER STATUS";
-        sql.set(showMasterStmt);
-        mysql.query(sql.get(), rs -> {
-            if (rs.next()) {
-                String binlogFilename = rs.getString(1);
-                long binlogPosition = rs.getLong(2);
-                source.setBinlogStartPoint(binlogFilename, binlogPosition);
-                if (rs.getMetaData().getColumnCount() > 4) {
-                    // This column exists only in MySQL 5.6.5 or later ...
-                    String gtidSet = rs.getString(5);// GTID set, may be null, blank, or contain a GTID set
-                    source.setCompletedGtidSet(gtidSet);
-                    logger.info("\t using binlog '{}' at position '{}' and gtid '{}'", binlogFilename, binlogPosition,
-                                gtidSet);
+        if (context.isSchemaOnlyRecoverySnapshot()) {
+            // We are in schema only recovery mode, use the existing binlog position            
+            if (source.binlogFilename() == null || source.binlogFilename().isEmpty()) {
+                // would like to also verify binlog position exists, but it defaults to 0 which is technically valid
+                throw new IllegalStateException("Could not find existing binlog information while attempting schema only recovery snapshot");
+            }
+            source.startSnapshot();
```

Comment:
What if the binlog was rotated in-between? You would then re-execute the snapshot and subsequently fail because you cannot continue with binlog processing. Should not by such case checked early?

Reply:
@jpechane Any relation to https://issues.jboss.org/browse/DBZ-414?
If the binlog position has rotated off I guess it would snapshot the schema and then immediately fail. Are you suggesting adding an additional check here or somewhere else?

Reply:
Yes, I suggest to first check if the position we remember is available and fail even before snapshotting the schema.